router.go

```diff
@@ -148,6 +210,46 @@ func (rt *Router) ReloadRoutes() {
 	routesCountMetric.Set(float64(rt.mux.RouteCount()))
 }
 
+func (rt *Router) getCurrentMongoInstance(db *mgo.Database) MongoReplicaSetMember {
+	defer func() {
+		if r := recover(); r != nil {
+			logWarn(r)
+		}
+	}()
+
+	replicaSetStatus := bson.M{}
+
+	if err := db.Run("replSetGetStatus", &replicaSetStatus); err != nil {
+		panic(fmt.Sprintf("router: couldn't get replica set status from MongoDB, skipping update (error: %v)", err))
+	}
+
+	replicaSetStatusBytes, err := bson.Marshal(replicaSetStatus)
+	if err != nil {
+		panic(fmt.Sprintf("router: couldn't marshal replica set status from MongoDB, skipping update (error: %v)", err))
+	}
+
+	replicaSet := MongoReplicaSet{}
+	err = bson.Unmarshal(replicaSetStatusBytes, &replicaSet)
+	if err != nil {
+		panic(fmt.Sprintf("router: couldn't unmarshal replica set status from MongoDB, skipping update (error: %v)", err))
+	}
+
+	currentInstance := make([]MongoReplicaSetMember, 0)
+	for _, instance := range replicaSet.Members {
+		if instance.Current == true {
+			currentInstance = append(currentInstance, instance)
+		}
+	}
+
+	logDebug("router: MongoDB instances", currentInstance)
+
+	if len(currentInstance) != 1 {
+		panic(fmt.Sprintf("router: did not find exactly one current MongoDB instance, skipping update (current instances found: %d)", len(currentInstance)))
+	}
```

Comment:
Thanks for considering this edge case :) I wonder if it might be safer to make it log an error and just return the first one that it found though, rather than not updating at all. What do you reckon?

Reply:
That's a possibility, but if you do this you then open yourself up to other edge cases i.e. you used the timestamp from the instance you were connected to but might then get another set of routes from another instance that hasn't yet updated, meaning you'd be back into the race condition we talked about a while ago. This would be a problem during times of infrequent changes, so probably not highly likely, but worth thinking about.

Reply:
Not a big deal - feel free to take or leave this one :)